File: src/main/java/io/bloco/faker/components/PhoneNumber.java
package io.bloco.faker.components;

import io.bloco.faker.FakerComponent;
import io.bloco.faker.FakerData;

public class PhoneNumber extends FakerComponent {

    private static final int DEFAULT_EXTENSION_LENGHT = 4;

    public PhoneNumber(FakerData data) {
        super(data);
    }

    public String phoneNumber() {
        return numerify(parse(fetch("phone_number.formats")));
    }

    public String cellPhone() {
        return numerify(parse(fetch("cell_phone.formats")));
    }

    // US only
    public String areaCode() {
        return fetch("phone_number.area_code");
    }

    // US only
    public String exchangeCode() {
        return fetch("phone_number.exchange_code");
    }

    // US only
    public String subscriberNumber() {
        return subscriberNumber(DEFAULT_EXTENSION_LENGHT);
    }

    // US only
    public String subscriberNumber(int length) {
        return String.format("%0" + length + "d", randomHelper.numberByLength(length));
    }

    // US only
    public String extension() {
        return subscriberNumber();
    }

    // US only
    public String extension(int length) {
        return subscriberNumber(length);
    }
}

File: lib/src/main/kotlin/io/bloco/faker/components/PhoneNumber.kt
package io.bloco.faker.components

import io.bloco.faker.FakerComponent
import io.bloco.faker.FakerData

class PhoneNumber(data: FakerData) : FakerComponent(data) {
    fun phoneNumber(): String {
        return numerify(parse(fetch("phone_number.formats")))
    }

    fun cellPhone(): String {
        return numerify(parse(fetch("cell_phone.formats")))
    }

    // US only
    fun areaCode(): String {
        return fetch("phone_number.area_code")
    }

    // US only
    fun exchangeCode(): String {
        return fetch("phone_number.exchange_code")
    }

    // US only
    fun subscriberNumber(): String {
        return subscriberNumber(DEFAULT_EXTENSION_LENGHT)
    }

    // US only
    fun subscriberNumber(length: Int): String {
        return String.format("%0" + length + "d", randomHelper.numberByLength(length))
    }

    // US only
    fun extension(): String {
        return subscriberNumber()
    }

    // US only
    fun extension(length: Int): String {
        return subscriberNumber(length)
    }

    companion object {
        private const val DEFAULT_EXTENSION_LENGHT = 4
    }
}